src/catalog/dependency_manager.cpp

```diff
@@ -433,17 +433,75 @@ string DependencyManager::CollectDependents(CatalogTransaction transaction, cata
 	return result;
 }
 
-void DependencyManager::DropObject(CatalogTransaction transaction, CatalogEntry &object, bool cascade) {
+void DependencyManager::VerifyExistence(CatalogTransaction transaction, DependencyEntry &object) {
+	auto &subject = object.Subject();
+
+	CatalogEntryInfo info;
+	if (subject.flags.IsOwnership()) {
+		info = object.SourceInfo();
+	} else {
+		info = object.EntryInfo();
+	}
+
+	auto &type = info.type;
+	auto &schema = info.schema;
+	auto &name = info.name;
+
+	auto &duck_catalog = catalog.Cast<DuckCatalog>();
+	auto &schema_catalog_set = duck_catalog.GetSchemaCatalogSet();
+
+	CatalogSet::EntryLookup lookup_result;
+	lookup_result = schema_catalog_set.GetEntryDetailed(transaction, schema);
+
+	if (type != CatalogType::SCHEMA_ENTRY && lookup_result.result) {
+		auto &schema_entry = lookup_result.result->Cast<SchemaCatalogEntry>();
+		lookup_result = schema_entry.GetEntryDetailed(transaction, type, name);
+	}
+
+	if (lookup_result.reason == CatalogSet::EntryLookup::FailureReason::DELETED) {
+		throw DependencyException("Could not commit creation of dependency, subject \"%s\" has been deleted",
+		                          object.SourceInfo().name);
+	}
+}
+
+void DependencyManager::VerifyCommitDrop(CatalogTransaction transaction, transaction_t start_time,
+                                         CatalogEntry &object) {
 	if (IsSystemEntry(object)) {
-		// Don't do anything for this
 		return;
 	}
-
 	auto info = GetLookupProperties(object);
-	// Check if there are any entries that block the DROP because they still depend on the object
-	catalog_entry_set_t to_drop;
+	ScanDependents(transaction, info, [&](DependencyEntry &dep) {
+		auto dep_committed_at = dep.timestamp.load();
+		if (dep_committed_at > start_time) {
```

Comment:
Do we need the `start_time` here? We should not have any dependencies at this point anymore, no?

Reply:
The drop of the dependents has not committed yet, so we need this check
I have added clarifying comments for this